File: 프로그래머스/lv1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.java
class Solution {
  public String[] solution(int n, int[] arr1, int[] arr2) {
      String[] answer = new String[n];
      String tmp;

      for(int i = 0 ; i < n ; i++){
          tmp = String.format("%16s", Integer.toBinaryString(arr1[i] | arr2[i]));
          answer[i] = tmp.substring(tmp.length() - n).replaceAll("1", "#").replaceAll("0", " ");
      }
      return answer;
  }
}

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.js
function solution(n, arr1, arr2) {
    var answer = [];
    for(let i=0; i<n; i++) {
        let bin1 = arr1[i].toString(2);
        bin1 = '0'.repeat(n - bin1.length) + bin1;
        
        let bin2 = arr2[i].toString(2);
        bin2 = '0'.repeat(n - bin2.length) + bin2;
        
        let str = ''
        
        for(let j=0; j<n; j++) {
            (bin1[j] === '1' || bin2[j] === '1') ? str+='#' : str+=' '
        }
        answer.push(str)
    }
    return answer;
}